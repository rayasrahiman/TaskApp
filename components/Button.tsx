import React, {CSSProperties} from 'react';
import {View, StyleSheet, Text, TouchableOpacity} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import {Colors} from '../constants/colors';

export type Props = {
  title: string;
  onPress(): void;
  btnWrapper: any;
  btnContainerProp: any;
  btnTxtProp: any;
};

const Button: React.FC<Props> = ({
  btnWrapper,
  title,
  onPress,
  btnContainerProp,
  btnTxtProp,
  ...touchable
}) => {
  return (
    <View style={{...styles.container, ...btnWrapper}}>
      <TouchableOpacity
        {...touchable}
        onPress={onPress}
        style={{...styles.buttonContainer, ...btnContainerProp}}>
        <Text style={{...styles.btnTxt, ...btnTxtProp}}>{title}</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginHorizontal: wp(5),
  },
  buttonContainer: {
    width: '100%',
    height: hp(7),
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.primary700,
  },
  btnTxt: {
    fontSize: hp(3),
    paddingVertical: hp(0.5),
    color: 'white',
  },
});

export default Button;
