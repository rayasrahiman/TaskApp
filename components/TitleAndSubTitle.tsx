import {View, Text, StyleSheet, Platform} from 'react-native';
import React from 'react';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';

import {Colors} from '../constants/colors';

export type Props = {
  title: string;
  subTitle: string;
  titleContProp: any;
  titleTxtProp: any;
  subTitleContProp: any;
  emailSubTitleContProp: any;
  subTitleTxtProp: any;
  viewPager: boolean;
  email: boolean;
  emailVal: string;
};

const TitleAndSubTitle: React.FC<Props> = ({
  title,
  subTitle,
  titleContProp,
  titleTxtProp,
  subTitleContProp,
  emailSubTitleContProp,
  subTitleTxtProp,
  viewPager,
  email,
  emailVal,
  ...titleAndSubTitleProp
}) => {
  return (
    <View {...titleAndSubTitleProp}>
      <View
        style={{
          ...{
            marginHorizontal: viewPager
              ? 0
              : Platform.OS === 'android'
              ? wp(5)
              : wp(3),
          },
          ...titleContProp,
        }}>
        <Text style={{...styles.titleTxt, ...titleTxtProp}}>{title}</Text>
      </View>
      <View
        style={{
          ...{
            marginHorizontal: viewPager
              ? 0
              : Platform.OS === 'android'
              ? wp(5)
              : wp(3),
          },
          ...subTitleContProp,
        }}>
        <Text
          style={{
            ...styles.subTitleTxt,
            ...{color: Colors.grayishBlack},
            ...subTitleTxtProp,
          }}>
          {subTitle}
        </Text>
      </View>
      {email && (
        <View
          style={{
            ...{
              marginHorizontal: viewPager
                ? 0
                : Platform.OS === 'android'
                ? wp(5)
                : wp(3),
              marginBottom: hp(2),
            },
            ...emailSubTitleContProp,
          }}>
          <Text
            style={{
              ...styles.subTitleTxt,
              ...{color: Colors.black},
              ...subTitleTxtProp,
            }}>
            {emailVal}
          </Text>
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  titleTxt: {
    color: Colors.darkBlue,
    fontWeight: 'bold',
    fontSize: Platform.OS === 'android' ? hp(5) : hp(4),
  },
  subTitleTxt: {
    color: Colors.grayishBlack,
    fontWeight: 'bold',
    fontSize: Platform.OS === 'android' ? hp(3) : hp(2),
  },
});

export default TitleAndSubTitle;
