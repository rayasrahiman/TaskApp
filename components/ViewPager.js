import React, {useState, useRef, useEffect} from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  FlatList,
  StyleSheet,
  Dimensions,
  Image,
} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';

import {Colors} from '../constants/colors';
import Button from './Button';
import TitleAndSubTitle from './TitleAndSubTitle';
import I18n from '../languages/i18n';

export default function ViewPager({
  data,
  horizontalIndicator,
  pagingEnabled,
  horizontal,
  BottomRgtBtn,
  navigation,
  ...fLprops
}) {
  const [imageActive, setImageActive] = useState(0);
  const ref = useRef(null);

  const onChange = e => {
    const contentOffsetX = e.nativeEvent.contentOffset.x;
    const currentIndex = Math.round(
      contentOffsetX / Dimensions.get('window').width,
    );
    setImageActive(currentIndex);
  };

  const goToNextSlide = () => {
    const nextSlideIndex = imageActive + 1;
    if (nextSlideIndex != data.length) {
      const offset = nextSlideIndex * Dimensions.get('window').width;
      ref?.current.scrollToOffset({offset});
      setImageActive(imageActive + 1);
    }
  };

  useEffect(() => {
    const interval = setInterval(() => {
      goToNextSlide();
    }, 5000);
    return () => clearInterval(interval);
  }, [imageActive]);

  const renderItem = ({item}) => {
    return (
      <View>
        <View style={styles.wrap}>
          <Image style={styles.wrap} source={item.img} resizeMode="cover" />
        </View>
        <View>
          <Image
            style={styles.img}
            resizeMode="contain"
            source={require('../assets/Logo.png')}
          />
        </View>
        <TitleAndSubTitle
          title={item.title}
          subTitle={item.subtitle}
          viewPager={true}
          titleTxtProp={styles.titleTxt}
          subTitleContProp={styles.subtitleCont}
          subTitleTxtProp={styles.subtitleTxt}
        />
      </View>
    );
  };

  return (
    <View style={styles.container}>
      <View>
        <FlatList
          {...fLprops}
          data={data}
          ref={ref}
          onMomentumScrollEnd={onChange}
          showsHorizontalScrollIndicator={horizontalIndicator}
          pagingEnabled={pagingEnabled}
          horizontal={horizontal}
          keyExtractor={(item, index) => item.id}
          renderItem={renderItem}
        />
      </View>
      {imageActive === data.length - 1 ? (
        <Button title={I18n.t('Continue')} onPress={navigation} />
      ) : (
        <View style={styles.bottomWrapper}>
          <View
            style={[
              styles.wrapDot,
              {
                flexDirection: 'row',
              },
            ]}>
            {data.map((e, index) => (
              <View
                key={e.id}
                style={imageActive === index ? styles.dotActive : styles.dot}
              />
            ))}
          </View>
          <TouchableOpacity onPress={navigation} style={styles.wrapDot}>
            <Text style={styles.btnTxt}>{BottomRgtBtn}</Text>
          </TouchableOpacity>
        </View>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.white,
  },
  wrap: {
    width: Dimensions.get('window').width,
    height: hp(50),
  },
  img: {
    height: hp(15),
    width: wp(15),
  },
  titleTxt: {
    marginHorizontal: hp(1),
    color: Colors.darkBlue,
    fontSize: hp(3.5),
    fontWeight: 'bold',
  },
  subtitleCont: {
    width: Dimensions.get('window').width,
  },
  subtitleTxt: {
    marginVertical: hp(5),
    marginHorizontal: hp(1),
    color: Colors.grayishBlack,
    fontSize: hp(3),
  },
  bottomWrapper: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    position: 'relative',
    alignItems: 'flex-end',
  },
  wrapDot: {
    marginHorizontal: wp(1),
    marginBottom: hp(2),
  },
  dotActive: {
    margin: 3,
    backgroundColor: Colors.darkBlue,
    borderRadius: hp(2.5) / 2,
    borderWidth: 5,
    borderColor: Colors.darkBlue,
    height: hp(2.5),
    width: hp(2.5),
  },
  dot: {
    margin: 3,
    backgroundColor: Colors.grayishBlack,
    borderRadius: hp(2.5) / 2,
    borderWidth: 5,
    borderColor: Colors.grayishBlack,
    height: hp(2.5),
    width: hp(2.5),
  },
  btnTxt: {
    fontSize: hp(3),
    color: Colors.grayishBlack,
  },
});
