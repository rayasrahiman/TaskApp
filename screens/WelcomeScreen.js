import React from 'react';

import {Images} from '../assets/Images';
import ViewPager from '../components/ViewPager';
import I18n from '../languages/i18n';

const WelcomeScreen = ({navigation}) => {
  return (
    <ViewPager
      testID="viewPager"
      data={Images}
      horizontalIndicator={false}
      pagingEnabled={true}
      horizontal={true}
      BottomRgtBtn={I18n.t('Skip')}
      navigation={() => navigation.navigate('Registration')}
    />
  );
};

export default WelcomeScreen;
