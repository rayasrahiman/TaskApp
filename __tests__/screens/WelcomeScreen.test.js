import React from 'react';
import {render} from '@testing-library/react-native';
import {shallow} from 'enzyme';

import WelcomeScreen from '../../screens/WelcomeScreen';
import ViewPager from '../../components/ViewPager';
import TitleAndSubTitle from '../../components/TitleAndSubTitle';
import Button from '../../components/Button';

describe('WelcomeScreen', () => {
  test('should render Welcome screen correctly', () => {
    const wrapper = render(<WelcomeScreen />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render ViewPager component', () => {
    const wrapper = shallow(<WelcomeScreen />);
    wrapper.find(<ViewPager />);
  });

  test('should render TitleAndSubTitle component', () => {
    const wrapper = shallow(<WelcomeScreen />);
    wrapper.find(<TitleAndSubTitle />);
  });

  test('should render Button component', () => {
    const wrapper = shallow(<WelcomeScreen />);
    wrapper.find(<Button />);
  });

  test('should render TitleAndSubTitle and other texts', () => {
    const wrapper = render(<WelcomeScreen />);
    wrapper.getByText("Take Control of your home's water.");
    wrapper.getByText('Stay smart connected and protected.');
    wrapper.getByText('Manage systems on the go.');
    wrapper.getByText('Get status information, home or away 24/7.');
    wrapper.getByText("We'll let you know what to do, when.");
    wrapper.getByText(
      'Receive alerts sent to your phone and if you choose your trusted pros.',
    );
    wrapper.getByText('Worry a lot less, enjoy a lot more.');
    wrapper.getByText('Peace of mins at your fingretips.');
    wrapper.getByText('Skip');
  });
});
