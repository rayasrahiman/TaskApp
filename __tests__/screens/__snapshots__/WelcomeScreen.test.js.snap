// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen should render Welcome screen correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View>
    <RCTScrollView
      data={
        Array [
          Object {
            "id": 1,
            "img": Object {
              "testUri": "../../../assets/img1.jpg",
            },
            "subtitle": "Stay smart connected and protected.",
            "title": "Take Control of your home's water.",
          },
          Object {
            "id": 2,
            "img": Object {
              "testUri": "../../../assets/img2.jpg",
            },
            "subtitle": "Get status information, home or away 24/7.",
            "title": "Manage systems on the go.",
          },
          Object {
            "id": 3,
            "img": Object {
              "testUri": "../../../assets/img3.jpg",
            },
            "subtitle": "Receive alerts sent to your phone and if you choose your trusted pros.",
            "title": "We'll let you know what to do, when.",
          },
          Object {
            "id": 4,
            "img": Object {
              "testUri": "../../../assets/img4.jpg",
            },
            "subtitle": "Peace of mins at your fingretips.",
            "title": "Worry a lot less, enjoy a lot more.",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      testID="viewPager"
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View>
            <View
              style={
                Object {
                  "height": 667,
                  "width": 750,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../assets/img1.jpg",
                  }
                }
                style={
                  Object {
                    "height": 667,
                    "width": 750,
                  }
                }
              />
            </View>
            <View>
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../assets/Logo.png",
                  }
                }
                style={
                  Object {
                    "height": 200,
                    "width": 112.5,
                  }
                }
              />
            </View>
            <View>
              <View
                style={
                  Object {
                    "marginHorizontal": 0,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#244379",
                      "fontSize": 46.5,
                      "fontWeight": "bold",
                      "marginHorizontal": 13.5,
                    }
                  }
                >
                  Take Control of your home's water.
                </Text>
              </View>
              <View
                style={
                  Object {
                    "marginHorizontal": 0,
                    "width": 750,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#838383",
                      "fontSize": 40,
                      "fontWeight": "bold",
                      "marginHorizontal": 13.5,
                      "marginVertical": 66.5,
                    }
                  }
                >
                  Stay smart connected and protected.
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View>
            <View
              style={
                Object {
                  "height": 667,
                  "width": 750,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../assets/img2.jpg",
                  }
                }
                style={
                  Object {
                    "height": 667,
                    "width": 750,
                  }
                }
              />
            </View>
            <View>
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../assets/Logo.png",
                  }
                }
                style={
                  Object {
                    "height": 200,
                    "width": 112.5,
                  }
                }
              />
            </View>
            <View>
              <View
                style={
                  Object {
                    "marginHorizontal": 0,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#244379",
                      "fontSize": 46.5,
                      "fontWeight": "bold",
                      "marginHorizontal": 13.5,
                    }
                  }
                >
                  Manage systems on the go.
                </Text>
              </View>
              <View
                style={
                  Object {
                    "marginHorizontal": 0,
                    "width": 750,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#838383",
                      "fontSize": 40,
                      "fontWeight": "bold",
                      "marginHorizontal": 13.5,
                      "marginVertical": 66.5,
                    }
                  }
                >
                  Get status information, home or away 24/7.
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View>
            <View
              style={
                Object {
                  "height": 667,
                  "width": 750,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../assets/img3.jpg",
                  }
                }
                style={
                  Object {
                    "height": 667,
                    "width": 750,
                  }
                }
              />
            </View>
            <View>
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../assets/Logo.png",
                  }
                }
                style={
                  Object {
                    "height": 200,
                    "width": 112.5,
                  }
                }
              />
            </View>
            <View>
              <View
                style={
                  Object {
                    "marginHorizontal": 0,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#244379",
                      "fontSize": 46.5,
                      "fontWeight": "bold",
                      "marginHorizontal": 13.5,
                    }
                  }
                >
                  We'll let you know what to do, when.
                </Text>
              </View>
              <View
                style={
                  Object {
                    "marginHorizontal": 0,
                    "width": 750,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#838383",
                      "fontSize": 40,
                      "fontWeight": "bold",
                      "marginHorizontal": 13.5,
                      "marginVertical": 66.5,
                    }
                  }
                >
                  Receive alerts sent to your phone and if you choose your trusted pros.
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View>
            <View
              style={
                Object {
                  "height": 667,
                  "width": 750,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../assets/img4.jpg",
                  }
                }
                style={
                  Object {
                    "height": 667,
                    "width": 750,
                  }
                }
              />
            </View>
            <View>
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../assets/Logo.png",
                  }
                }
                style={
                  Object {
                    "height": 200,
                    "width": 112.5,
                  }
                }
              />
            </View>
            <View>
              <View
                style={
                  Object {
                    "marginHorizontal": 0,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#244379",
                      "fontSize": 46.5,
                      "fontWeight": "bold",
                      "marginHorizontal": 13.5,
                    }
                  }
                >
                  Worry a lot less, enjoy a lot more.
                </Text>
              </View>
              <View
                style={
                  Object {
                    "marginHorizontal": 0,
                    "width": 750,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#838383",
                      "fontSize": 40,
                      "fontWeight": "bold",
                      "marginHorizontal": 13.5,
                      "marginVertical": 66.5,
                    }
                  }
                >
                  Peace of mins at your fingretips.
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "position": "relative",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginBottom": 26.5,
            "marginHorizontal": 7.5,
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#244379",
            "borderColor": "#244379",
            "borderRadius": 16.75,
            "borderWidth": 5,
            "height": 33.5,
            "margin": 3,
            "width": 33.5,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#838383",
            "borderColor": "#838383",
            "borderRadius": 16.75,
            "borderWidth": 5,
            "height": 33.5,
            "margin": 3,
            "width": 33.5,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#838383",
            "borderColor": "#838383",
            "borderRadius": 16.75,
            "borderWidth": 5,
            "height": 33.5,
            "margin": 3,
            "width": 33.5,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#838383",
            "borderColor": "#838383",
            "borderRadius": 16.75,
            "borderWidth": 5,
            "height": 33.5,
            "margin": 3,
            "width": 33.5,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginBottom": 26.5,
          "marginHorizontal": 7.5,
          "opacity": 1,
        }
      }
      testID="btnRight"
    >
      <Text
        style={
          Object {
            "color": "#838383",
            "fontSize": 40,
          }
        }
      >
        Skip
      </Text>
    </View>
  </View>
</View>
`;
